### PR TITLE
One instance of objectMapper, typed RequestBody

### PR DESCRIPTION
**Pull request by svenngjeto**

### CODE REVIEW

File: apigateway/src/test/java/nva/commons/apigateway/testutils/Handler.java [hunk 1]
```diff
@@ -39,22 +39,22 @@ public Handler(Environment environment, ObjectMapper mapper) {
...
-    protected String processInput(RequestBody input, RequestInfo requestInfo, Context context)
+    protected RequestBody processInput(RequestBody input, RequestInfo requestInfo, Context context)
         throws ApiGatewayException {
         this.headers = requestInfo.getHeaders();
         this.proxy = requestInfo.getPathParameters().get(PROXY_TAG);
         this.path = requestInfo.getPath();
         this.body = input;
         this.addAdditionalHeaders(() -> additionalHeaders(body));
-        return String.join(",", input.getField1(), input.getField2());
+        return this.body; //String.join(",", input.getField1(), input.getField2());
```
Comment (axthosarouris) [inline apigateway/src/test/java/nva/commons/apigateway/testutils/Handler.java:49]: just delete the old code, no need to comment it out.

File: core/src/test/java/nva/commons/core/JsonUtilsTest.java [hunk 2]
```diff
@@ -120,15 +114,15 @@ public void objectMapperSerializesEmptyStringAsNull() throws JsonProcessingExcep
         mapToSerialize.put("nullValue", null);
         String actualJson = objectMapper.writeValueAsString(mapToSerialize);
         ObjectNode sampleJsonObjectWithoutValue = objectMapper.createObjectNode();
-        String expectedJson = objectMapper.writeValueAsString(sampleJsonObjectWithoutValue);
+        String expectedJson = OBJECT_WITH_NULL_VALUES;
```
Comment (axthosarouris) [inline core/src/test/java/nva/commons/core/JsonUtilsTest.java:117]: Use the class `SamplePojo` and the resource files in the folder `jsonutils`. What we want here is that when we de-serialize objects the empty strings are initlialized as null. This test is testing the serialization.

**Review by axthosarouris — CHANGES_REQUESTED** [review]

File: apigateway/src/test/java/nva/commons/apigateway/testutils/Handler.java (hunk 1 above)
Comment (svenngjeto) [reply apigateway/src/test/java/nva/commons/apigateway/testutils/Handler.java]: I guess thats why we use version control.... :-)

File: core/src/test/java/nva/commons/core/JsonUtilsTest.java (hunk 2 above)
Comment (svenngjeto) [reply core/src/test/java/nva/commons/core/JsonUtilsTest.java]: Tests rewritten

File: core/src/test/java/nva/commons/core/JsonUtilsTest.java [hunk 3]
```diff
@@ -113,22 +108,28 @@ public void objectMapperSerializesInstantAsIsoStrings() throws JsonProcessingExc
...
+
     @Test
     public void objectMapperSerializesEmptyStringAsNull() throws JsonProcessingException {
-        Map<String, Object> mapToSerialize = new HashMap<>();
-        mapToSerialize.put("emptyString", "");
-        mapToSerialize.put("nullValue", null);
-        String actualJson = objectMapper.writeValueAsString(mapToSerialize);
-        ObjectNode sampleJsonObjectWithoutValue = objectMapper.createObjectNode();
-        String expectedJson = objectMapper.writeValueAsString(sampleJsonObjectWithoutValue);
+        String expectedJson = IoUtils.stringFromResources(Path.of(JSON_UTILS_RESOURCES, POJO_WITH_MISSING_VALUES));
```
Comment (axthosarouris) [inline core/src/test/java/nva/commons/core/JsonUtilsTest.java:119]: I think what we need here is to test the de-serialization:
```
@ParameterizedTest
    @ValueSource(strings={POJO_WITH_MISSING_VALUES, POJO_WITH_EMPTY_VALUES})
    public void objectMapperDeserializesEmptyStringAsNull(String resourceInput) throws JsonProcessingException {
        String jsonString = IoUtils.stringFromResources(Path.of(JSON_UTILS_RESOURCES,resourceInput));
        SamplePojo actualObject = objectMapper.readValue(jsonString,SamplePojo.class);
        assertThat(actualObject.getField2(),is(nullValue()));
        assertThat(actualObject.getField1(),is(not(nullValue())));
    }
```

File: core/src/test/java/nva/commons/core/JsonUtilsTest.java [hunk 4]
```diff
@@ -113,22 +108,29 @@ public void objectMapperSerializesInstantAsIsoStrings() throws JsonProcessingExc
         assertThat(jsonNow, is(equalTo(expectedString)));
     }
 
+    @Test
+    public void objectMapperSerializesNullStringAsNull() throws JsonProcessingException {
```
Comment (axthosarouris) [inline core/src/test/java/nva/commons/core/JsonUtilsTest.java:110]: I think this test is covered by the one in the comment below

**Review by axthosarouris — CHANGES_REQUESTED** [review]

**Review by axthosarouris — DISMISSED** [review]

**Review by axthosarouris — APPROVED** [review]

**Pull request merged**